### ISSUE TITLE
Safely getting at values

### ISSUE THREAD

**Issue by eh-am** [true-myth/true-myth#271]
Hello :wave:, maybe I am doing something very wrong, but [the "Safely getting at values" example from README](https://github.com/true-myth/true-myth#safely-getting-at-values) isn't working properly since 5.0.0:
![image](https://user-images.githubusercontent.com/6951209/147279817-866d8c92-0e8c-4022-947d-b5c832ad8a40.png)

(Line 7 should've triggered an error)

But it works on 4.1.1
![image](https://user-images.githubusercontent.com/6951209/147279629-b7d2b756-e4c6-47af-8bf3-68f0866e00b2.png)

**Comment by chriskrycho** [comment]
Fix released in v5.1.2.

**Comment by eh-am** [comment]
Hey, it's me again. I have a similar problem with `Result`, now:
```js
 function mightSucceed(doesSucceed: boolean): Result<number, string> {                                                             
   return doesSucceed ? Result.ok(42) : Result.err('something went wrong!');                                         
 }
                                                                                                                                                                                                                                             
const a = mightSucceed(false);                                                                                      
a.value; // should've errored
```

Repo
https://github.com/eh-am/true-myth-inferr-type-error/blob/497f2acae567cf2b341a3047a84db6614f0f45ec/index.ts
